src/aind_behavior_curriculum/tmp.py

```diff
@@ -0,0 +1,80 @@
+"""
+Work in Progress Curriculum
+"""
+
+from collections import defaultdict
+
+class Task:
+    """
+    Defined in exisiting PR
+    """
+    pass
+
+class Metric:
+    """
+    Initalizes a connection to relevant database (referencing some configuration file)
+    User extends this with their own metric implementation based on data inside the database.
+    """
+    pass
+
+
+class Evaluation:
+    """
+    Serializable Callable, takes in a list of metrics that is serialized alongside.
+    """
+    pass
+
+
+# TODO: De/serialization
+class Curriculum:
```

Comment:
I think we will need pop/del methods too. For instance, if you pop a stage you want to make sure you tell the user that some connections will have to be removed too. Also if you add a transition one should check if the stages are available too. 

For the transitions we will also need methods to order the evaluation of the rules. Since the order matters